x/feegrant/filtered_fee.go

```diff
@@ -61,7 +61,14 @@ func (a *AllowedMsgAllowance) Accept(ctx sdk.Context, fee sdk.Coins, msgs []sdk.
 		return false, err
 	}
 
-	return allowance.Accept(ctx, fee, msgs)
+	remove, reserr := allowance.Accept(ctx, fee, msgs)
+	if !remove {
+		a.Allowance, err = types.NewAnyWithValue(allowance.(proto.Message))
```

Comment:
Assign the changed allowance to the old one, so the change could be applied.
There is no need to do this job if `remove` is `true` apparently.